### PR TITLE
Review transaction execution UI

### PR DESCRIPTION
**Pull request by elgatovital**
Handles #1933

Changes proposed in this pull request:
- Load chain l2 flag and features
- DB migration
- Add review transaction execution UI
- Adjust tx details & enable execution button
- Extend nav graph
- Cusom view for select key item
- Loading native token balances via rpc
- abi definitions for different safe contract versions

Not part of this PR: estimation & submission

@gnosis/mobile-devs

### CODE REVIEW

**Comment by biafra23** [comment]
Not sure this is part of this PR. But the submit button on the  execute transaction screen is not enabled. Even though an owner key with sufficient funds is imported. 
image

**Comment by elgatovital** [comment]
> Not sure this is part of this PR. But the submit button on the execute transaction screen is not enabled. Even though an owner key with sufficient funds is imported. 
execution itself is not part of this pr. Submit button will remain disabled.

File: app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModel.kt [hunk 1]
```diff
@@ -94,6 +101,14 @@ class TransactionDetailsViewModel
                 }
     }
 
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    suspend fun canBeExecutedFromDevice(
+        executionInfo: DetailedExecutionInfo.MultisigExecutionDetails,
+        localOwners: List<Owner>
+    ): Boolean {
+        return localOwners.isNotEmpty() && executionInfo.confirmations.size == executionInfo.confirmationsRequired
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModel.kt:109]: We're not supposed to check the balances here as well?

**Review by biafra23 — APPROVED** [review]
LGTM

**Pull request merged**